elastic/kibana
# [Discover][New table] Fix selected documents no longer in the sample of 500 documents

## Summary

In the EuiDataGrid document table there is a document selection offered. This was meant to restore a basic comparison functionality like in the legacy grid. The problem with the current implementation is, that this selection is currently based on the sample of 500 documents that Discover fetches from ES. The ids of the documents selected are stored in the UI, but not the records. So it's implemented like a client side filter. If the underlying data of 500 changes (different time range, sorting, filter, ...), and those don't contain the selected records, the selection needs to change, since else it would display an empty list if "Show only selected records" is selected:

![](https://user-images.githubusercontent.com/6256687/116923325-7ebc4580-ac24-11eb-9a96-28bfb98f2518.png)

With this PR in this case all available records are displayed, and the "Show selected documents only" mode is disabled.

### Testing

* Enable `New table` in `Advanced settings`
* Go to Discover, select one of our sample sets
* Choose a long time range so it contains more than 500 records 
* Select a few documents 
* Select "Show selected documents only", now you should see only the selected documents
* Invert sorting of the documents
* Now you should no longer view an empty table like before this PR, but the table filled with unselected documents

Note that this in an initial fix, it's not a good user experience. In a further step we need to assure that in this case the selection is still displayed, just the sorting of the selected documents changed. But for this we need to implement a query sent to ES for retrieving the selected records, so this is a follow up issue #99775.

Fixes #99105

### Checklist
